deploy/minikube/Start-Minikube.ps1

```diff
@@ -0,0 +1,98 @@
+# Import config
+$config = "$PSScriptRoot\minikube_config.ps1"
+if (!(Test-Path $config)) {
+    Throw "$config was not found"
+}
+. $config
+
+# Test if VM is Running
+$vm = Get-VM |
+    Where-Object { $_.Name -eq $config_minikube_vm_name }
+if ($vm) {
+    if ($vm.State -ne "off") {
+        Write-Host -f "Cyan" -Object "$config_minikube_vm_name is already running"
+        & "$PSScriptRoot\Share-HostDrive.ps1"
+        return
+    }
+}
+
+# Install minikube
+if (! (Test-Path $config_minikube)) {
+    Write-Host -f "Cyan" -Object "Downloading minikube.exe"
+    $minikube_url = "https://storage.googleapis.com/minikube/releases/latest/minikube-windows-amd64.exe"
+    $web_client = New-Object System.Net.WebClient
+    $web_client.DownloadFile($minikube_url, $config_minikube)
+    if ($?) {
+        Write-Host "minikube.exe downloaded`n"
+    } else {
+        Throw "Unable to download minikube from $minikube_url"
+    }
+}
+
+# Create Hyper-V switch
+$ethernet_adapter_name = "Ethernet"
+$ethernet_adapter = Get-NetAdapter -Name $ethernet_adapter_name
+$external_vm_switchs = Get-VMSwitch |
+    Where-Object {
+        $_.SwitchType -eq "external" `
+        -and $ethernet_adapter.InterfaceDescription -eq $_.NetAdapterInterfaceDescription
+    }
+
+if (!$external_vm_switchs) {
+    Write-Host -f "Cyan" -Object "Adding external VMSwitch"
+    $external_vm_switchs = New-VMSwitch `
+        -Name $config_minikube_vmswitch_name `
+        -NetAdapterName $ethernet_adapter_name `
+        -AllowManagementOS $true `
+        -Notes "VMSwitch for Minikube with ReactiveTrader"
+    if ($?) {
+        Write-Host -Object "VMSwitch created`n"
+    }
+    else {
+        Throw "Unable to create external VMSwitch for minikube"
+    }
+}
+
+# Remove kubectl message
+& $config_minikube config set WantKubectlDownloadMsg false
+
+# start
+Write-Host -f "Cyan" "Starting minikube:"
+& $config_minikube start `
+    --vm-driver="hyperv" `
+    --hyperv-virtual-switch="$($external_vm_switchs.Name)" `
+    --memory 6144
```

Comment:
also prefer to lower this if possible - perhaps query the current host free memory amount first?